yarl/_url.py

```diff
@@ -925,6 +949,42 @@ def _make_netloc(
             ret = user + "@" + ret
         return ret
 
+    @classmethod
+    @lru_cache  # match the same size as urlsplit
+    def _split_netloc(
+        cls,
+        netloc: str,
+    ) -> Tuple[Optional[str], Optional[str], Optional[str], Optional[int]]:
+        """Split netloc into username, password, host and port."""
+        if "@" not in netloc:
+            username: Optional[str] = None
+            password: Optional[str] = None
+            hostinfo = netloc
+        else:
+            userinfo, _, hostinfo = netloc.rpartition("@")
+            username, have_password, password = userinfo.partition(":")
+            if not have_password:
+                password = None
+
+        if "[" in hostinfo:
+            _, _, bracketed = hostinfo.partition("[")
+            hostname, _, port_str = bracketed.partition("]")
+            _, _, port_str = port_str.partition(":")
+        else:
+            hostname, _, port_str = hostinfo.partition(":")
+
+        if not port_str:
+            port: Optional[int] = None
```

Comment:
```suggestion
            port: Union[int, None] = None
```

Reply:
I'm so used to writing `Optional`. Let me do https://github.com/aio-libs/yarl/pull/1095 first so I won't add any more of these and remove `Optional` from the imports

Reply:
Would https://pypi.org/p/no-optional help?

Reply:
Once the import is gone I'll remember.

Reply:
Yeah, but in general, it might be better for other contributors. That's for the future, though.